### ISSUE TITLE
Space + Any key prefix + ? is not working.

### ISSUE THREAD

**Issue by zilongshanren** [syl20bnr/spacemacs#2844]
It is working before and I could use it to search all the related commands.

It's really handy to add the `SPC + KeyPrefix + ?` support.

**Comment by syl20bnr** [comment]
I may be due to which-key cc @justbur 

@justbur do you think it is possible to create a configurable passthrough for `?`, it could be valid in spacemacs at least where we can define a convention to preserve `?`  (actually I update a key binding recently because it made use of `?` which should be forbidden).

**Comment by justbur** [comment]
I don't see where this happens, but it looks like `?` is somehow equivalent `C-h`, which which-key uses by default for paging. The configuration variable is `which-key-use-C-h-for-paging`, which is easily disabled. I'll look into using `?` for `helm-descbinds`, which is the command that's currently shadowed.

Another alternative is to bind keys for paging in which-key and retain the old behavior with C-h.

**Comment by justbur** [comment]
@zilongshanren Are you sure this was the behavior before? I only see "?" bound in one place, [here](https://github.com/syl20bnr/spacemacs/blob/master/spacemacs/packages.el#L1982), and that's only for the root prefix, "SPC". What you say is true of `C-h`, but that one binding for `?` is not being affected for me.

**Comment by zilongshanren** [comment]
@justbur  
Yes, I'm 100% sure.
The `SPC + Key-Prefix + ?` is binded to `helm-descbinds`.
But now only root `SPC` prefix is supported.

**Comment by justbur** [comment]
So you can set `which-key-use-C-h-for-paging` to `nil` to prevent this from happening, but I think I have a nice solution to the problem. I'll introduce a companion variable that if true (which I think will be the default) which key will drop into `describe-prefix-bindings` after going past the last page (right now which-key just cycles through the pages).

**Comment by zilongshanren** [comment]
@justbur  
I have discussed it here :  https://github.com/syl20bnr/spacemacs/issues/2054

**Comment by zilongshanren** [comment]
@justbur  
I have enabled ` which-key-use-C-h-for-paging` to nil, but it is still not working.

**Comment by justbur** [comment]
Where did you set it? You either need to set it in `dotspacemacs/init` or toggle the mode off, set it then turn it back on.

**Comment by justbur** [comment]
I'm going to make a couple of changes to which-key now, which should make this problem all but go away.

**Comment by zilongshanren** [comment]
@justbur 
I set it in the `dotspacemacs/config` function and I also tried the `dotspacemacs/init` function.
Both are not working.

**Comment by justbur** [comment]
ok, I see the problem. Let me make a couple of changes which should fix this.

**Comment by justbur** [comment]
After #2847 is merged, you shouldn't have any more issues if you set that variable in your dotspacemacs/init function. 

I'm working on a better solution on the which-key side.

**Comment by justbur** [comment]
@syl20bnr I have what I think is a nice solution for this problem. Which-key now supports both paging with C-h and using helm-descbinds. If we set `which-key-prevent-C-h-from-cycling` to `t` then when there are multiple pages of keys `C-h` will page through them until the last page. Hitting `C-h` (or `?` which is equivalent) at this point will trigger the default behavior of calling `helm-descbinds` in the spacemacs case.

**Comment by syl20bnr** [comment]
@justbur very nice :+1:

**Comment by geo7** [comment]
Is this the same issue : 

#[http://sendvid.com/pgu9bhrq](http://sendvid.com/pgu9bhrq)

JavaScript file, update spacemacs commit : 

```
commit fba874d64d2c6eb3239b68d226a49936ecf27e4e
Author: syl20bnr <sylvain.benner@gmail.com>
Date:   Mon Aug 31 23:35:28 2015 -0400
```

If this is a separate issue I'll delete this and make a new one

**Comment by justbur** [comment]
@geo7 I can't replicate this, and `SPC m g g` looks like a command, not a prefix. Which-key doesn't show anything for commands. It looks like you don't have any keys bound for the prefix `SPC m g` which is why which-key says that. I just loaded the javascript layer and saw keys under `SPC m g` with which-key just fine.

And no that would not be the same issue as this one.
